### PR TITLE
ft/mobile-navigation

### LINKED ISSUE #18: Mobile Toolbar
Address issues raised on [review](https://github.com/CodeForAfrica/harassment-manager/pull/17#pullrequestreview-1169824319). Mainly:

> We can improve the pop-up to closely match the designs... especially the spacing between menu items.

S

And any new issues that will be discovered.

### PR DESCRIPTION
**Pull request by kelvinkipruto**
This PR implements mobile navigation.

Fixes #18 

## Screenshots

nav1
nav2

### CODE REVIEW

File: src/app/toolbar/toolbar.component.scss [hunk 1]
```diff
@@ -177,63 +177,62 @@
...
-      }
+    a {
+      width: 50%;
+      align-items: center;
+
     }
+
     /*Hide find support button on mobile since it already is in footer */
-    .find-support {
+    .opportunities {
```
Comment (kilemensi) [inline src/app/toolbar/toolbar.component.scss:226]: Opportunities link is not in the footer yet. Lets add it.

File: src/app/toolbar/toolbar.component.scss [hunk 2]
```diff
@@ -308,4 +433,4 @@
     left: calc((1144px - 100vw) / 2);
     right: calc((1144px - 100vw) / 2);
   }
-}
+}
```
Comment (kilemensi) [inline src/app/toolbar/toolbar.component.scss:436]:
```suggestion
}

```

**Review by kilemensi — COMMENTED** [review]
👍🏽 

----

We can improve the pop-up to closely match the designs... especially the spacing between menu items.

S

**Review by kilemensi — COMMENTED** [review]
QR: Can we show mobile toolbar in tablet view as well (768px)? The desktop toolbar doesn't work well here.

Screenshot from 2022-11-07 13-43-24

File: src/app/toolbar/toolbar.component.scss (hunk 2 above)
Comment (kilemensi) [reply src/app/toolbar/toolbar.component.scss:444]: This is not resolved.

**Comment by kelvinkipruto** [comment]
#18 Ready for rereview

**Review by kilemensi — COMMENTED** [review]
👍🏽 @kelvinkipruto 

----------------

The toolbar links alignment looks off on my end.

M1


And the pop up is missing current page indicator (and fonts are larger than in the designs).

M2

File: src/app/toolbar/toolbar.component.html [hunk 3]
```diff
@@ -15,21 +15,24 @@
...
-    <ng-container *ngIf="!userIsSignedIn">
-      <span class="spacer"></span>
-      <button mat-button class="action-button get-started-button" routerLink="/">Get Started</button>
-    </ng-container>
+
+
+    <button mat-button class="action-button mobile-menu-button" (click)="toggleMobileMenu()">
+      <mat-icon svgIcon="hamburger_menu"></mat-icon>
+    </button>
+
```
Comment (kilemensi) [inline src/app/toolbar/toolbar.component.html:28]:
```suggestion
    <button mat-button class="action-button mobile-menu-button" (click)="toggleMobileMenu()">
      <mat-icon svgIcon="hamburger_menu"></mat-icon>
    </button>
```

Why so many empty lines?

File: src/app/toolbar/toolbar.component.html [hunk 4]
```diff
@@ -15,21 +15,24 @@
...
   </mat-toolbar-row>
 
-  <mat-toolbar-row class="toolbar-row toolbar-row-signed-in" *ngIf="userIsSignedIn">
+  <mat-toolbar-row class="toolbar-row main-nav toolbar-row-signed-in" *ngIf="userIsSignedIn"
+    [ngClass]="isMobileMenuOpen ? 'hidden-nav': ''  ">
     <div class="nav-links nav-item">
       <a routerLinkActive="active-router-link" routerLink="/home" #homeLink role="link"
-        [attr.aria-current]="homeLink.routerLinkActive ? 'page' : false">
+        [attr.aria-current]="homeLink.routerLinkActive ? 'page' : false" class="home-link">
```
Comment (kilemensi) [inline src/app/toolbar/toolbar.component.html:35]:
```suggestion
        [attr.aria-current]="homeLink.routerLinkActive ? 'page' : false">
```
Are we using `home-link` anywhere?

File: src/app/toolbar/toolbar.component.html [hunk 5]
```diff
@@ -54,6 +57,40 @@
...
+    <div class="nav-links nav-item sign-out-nav" *ngIf="userIsSignedIn">
+      <button class="sign-out-button" (click)="revokeAuthorization()">
+        <span>Sign out</span>
+      </button>
+    </div>
+  </mat-toolbar-row>
+
+  <mat-toolbar-row class="toolbar-row toolbar-row-signed-in mobile-nav"
+    [ngClass]="isMobileMenuOpen ? 'mobile-menu-open': 'mobile-menu-closed' ">
+
```
Comment (kilemensi) [inline src/app/toolbar/toolbar.component.html:69]:
```suggestion
```

File: src/app/toolbar/toolbar.component.scss [hunk 6]
```diff
@@ -231,61 +231,66 @@ app-report-progress-bar {
   display: none;
 }
 
+
+
```
Comment (kilemensi) [inline src/app/toolbar/toolbar.component.scss:235]:
```suggestion
```

File: src/app/toolbar/toolbar.component.scss [hunk 7]
```diff
@@ -231,61 +231,66 @@ app-report-progress-bar {
...
-      height: 50%;
-      margin: 0 20px 0 0;
       padding: 0;
       text-decoration: none;
 
       img {
         display: none;
       }
 
+
```
Comment (kilemensi) [inline src/app/toolbar/toolbar.component.scss:265]:
```suggestion
```

File: src/app/toolbar/toolbar.component.scss [hunk 8]
```diff
@@ -231,61 +231,66 @@ app-report-progress-bar {
...
-        height: 29px;
-        width: 29px;
-        align-self: flex-end;
-        margin-bottom: -12px;
-      }
+      // 2nd and 5th child justifies content to the right
+      // &:nth-child(2),
+      // &:nth-child(5) {
+      //   justify-content: flex-end;
+      // }
```
Comment (kilemensi) [inline src/app/toolbar/toolbar.component.scss:282]: If we no longer need a piece of code, delete it.

File: src/app/toolbar/toolbar.component.scss [hunk 9]
```diff
@@ -301,6 +306,93 @@ app-report-progress-bar {
   top: 0;
 }
 
+.mobile-nav {
+  height: 100vh;
+  width: 100vw;
+  flex-direction: column;
+
+  // target the first child of the mobile nav
```
Comment (kilemensi) [inline src/app/toolbar/toolbar.component.scss:314]:
```suggestion
```
The code is self-explanatory.

File: src/app/toolbar/toolbar.component.scss [hunk 10]
```diff
@@ -301,6 +306,93 @@ app-report-progress-bar {
...
+  >div:first-child {
+    align-self: flex-start;
+    flex: .5;
+  }
+
+  // target only second div
```
Comment (kilemensi) [inline src/app/toolbar/toolbar.component.scss:320]:
```suggestion
```

File: src/app/toolbar/toolbar.component.scss [hunk 11]
```diff
@@ -301,6 +306,93 @@ app-report-progress-bar {
...
+  >div:nth-child(2) {
+    justify-content: center;
+    flex: 2;
+  }
+
+  // target third div
```
Comment (kilemensi) [inline src/app/toolbar/toolbar.component.scss:326]:
```suggestion
```

File: src/app/toolbar/toolbar.component.scss [hunk 12]
```diff
@@ -301,6 +306,93 @@ app-report-progress-bar {
...
+    justify-content: center;
+  }
+
+  .close-mobile-menu-button {
+    background-color: transparent;
+    align-self: center;
+    top: 40px;
+  }
+
+
```
Comment (kilemensi) [inline src/app/toolbar/toolbar.component.scss:338]:
```suggestion
```

File: src/app/toolbar/toolbar.component.scss [hunk 13]
```diff
@@ -301,6 +306,93 @@ app-report-progress-bar {
...
+  .nav-links {
+    flex-direction: column;
+    align-items: center;
+    justify-content: space-around;
+
+
```
Comment (kilemensi) [inline src/app/toolbar/toolbar.component.scss:344]:
```suggestion
```

File: src/app/toolbar/toolbar.component.scss [hunk 14]
```diff
@@ -301,6 +306,93 @@ app-report-progress-bar {
...
+      img {
+        display: block;
+        height: 29px;
+        width: 29px;
+        align-self: flex-end;
+        margin-bottom: -12px;
+      }
+    }
+
+
```
Comment (kilemensi) [inline src/app/toolbar/toolbar.component.scss:359]:
```suggestion
```

File: src/app/toolbar/toolbar.component.scss [hunk 15]
```diff
@@ -301,6 +306,93 @@ app-report-progress-bar {
...
+    a,
+    button {
+      span {
+        font-size: 15px;
+        line-height: 19px;
+        font-weight: 700;
+      }
+    }
+  }
+
```
Comment (kilemensi) [inline src/app/toolbar/toolbar.component.scss:377]:
```suggestion
```

File: src/app/toolbar/toolbar.component.scss [hunk 16]
```diff
@@ -311,16 +403,58 @@ app-report-progress-bar {
       padding-right: 0;
     }
 
-    .get-started-button {
-      display: initial;
+  }
+
+  .toolbar-row-signed-in {
+
```
Comment (kilemensi) [inline src/app/toolbar/toolbar.component.scss:409]:
```suggestion
```

File: src/app/toolbar/toolbar.component.scss [hunk 17]
```diff
@@ -311,16 +403,58 @@ app-report-progress-bar {
...
+        }
+      }
+
+      .opportunities,
+      .find-support {
+        display: none;
+      }
+
     }
+
```
Comment (kilemensi) [inline src/app/toolbar/toolbar.component.scss:446]:
```suggestion
    }
```

File: src/app/toolbar/toolbar.component.scss [hunk 18]
```diff
@@ -311,16 +403,58 @@ app-report-progress-bar {
...
 
   .toolbar-row-signed-in::before {
     left: calc((728px - 100vw) / 2);
     right: calc((728px - 100vw) / 2);
   }
 
+  .main-nav {
+    display: flex;
+  }
+
```
Comment (kilemensi) [inline src/app/toolbar/toolbar.component.scss:457]:
```suggestion
```

File: src/app/toolbar/toolbar.component.scss [hunk 19]
```diff
@@ -347,12 +481,27 @@ app-report-progress-bar {
...
 
   .toolbar-row-signed-in {
+    height: 80px;
+
+    >div:first-child {
+      flex: 6;
+      flex-wrap: nowrap;
+    }
+
+    // target second div
```
Comment (kilemensi) [inline src/app/toolbar/toolbar.component.scss:491]:
```suggestion
```

File: src/app/toolbar/toolbar.component.scss (hunk 2 above)
Comment (kilemensi) [reply src/app/toolbar/toolbar.component.scss:542]: When you see the red circle with a dash icon @kelvinkipruto, it means there is a missing new line at the end of a file. This is not resolved!

File: src/app/toolbar/toolbar.component.ts [hunk 20]
```diff
@@ -141,6 +148,7 @@ export class ToolbarComponent {
       this.sanitizer.bypassSecurityTrustResourceUrl('/close_icon.svg')
     );
 
+
```
Comment (kilemensi) [inline src/app/toolbar/toolbar.component.ts:151]:
```suggestion
```

File: src/app/toolbar/toolbar.component.scss [hunk 21]
```diff
@@ -119,9 +118,9 @@ app-report-progress-bar {
   background-color: #ffffff;
 }
 
-.get-started-button {
-  background-color: #101bcf;
-  display: none;
+.mobile-menu-button {
+  background-color: #eaefff;
+  color: #101bcf;
```
Comment (kilemensi) [inline src/app/toolbar/toolbar.component.scss:123]:
```suggestion
  color: #101bcf;
  min-width: 0;
  padding: 0;
```

**Review by kilemensi — APPROVED** [review]
:rocket:

**Pull request merged**